packages/trpc/server/routers/viewer/workflows.tsx

```diff
@@ -224,6 +226,40 @@ export const workflowsRouter = createProtectedRouter()
         },
       });
 
+      const newActiveEventTypes = activeOn.filter((eventType) => {
+        if (
+          !oldActiveOnEventTypes ||
+          !oldActiveOnEventTypes
+            .map((oldEventType) => {
+              return oldEventType.eventTypeId;
+            })
+            .includes(eventType)
+        ) {
+          return eventType;
+        }
+      });
+
+      //check if new event types belong to user
+      for (const newEventTypeId of newActiveEventTypes) {
+        const newEventType = await ctx.prisma.eventType.findFirst({
+          where: {
+            id: newEventTypeId
+          },
+          include: {
+            team: {
+              include: {
+                members: true,
+              }
+            }
+          }
+        })
+
+        if (newEventType && newEventType.userId !== user.id && newEventType?.team?.members.filter(membership => { membership.userId === user.id }).length) {
+          throw new TRPCError({ code: "UNAUTHORIZED" });
```

Comment:
Throw an error when a user tries to set an event type as active that does not belong to the user or a team where the user is member of